src/Controller/CRUDController.php

```diff
@@ -480,6 +481,9 @@ public function batchAction()
         }
 
         $datagrid = $this->admin->getDatagrid();
+        if (!($datagrid instanceof Datagrid)) {
```

Comment:
Maybe create a method like `getDatagrid` for this

Reply:
in the controller? or?

Reply:
ping @core23

Reply:
@core23 please give some feedback

Reply:
> in the controller? or?

Yes, maybe something like `CRUDController::getDatagrid` which does all the `build*` calls just once.